R/PredictionSurv.R

```diff
@@ -13,12 +13,22 @@
 #' learner = lrn("surv.kaplan")
 #' p = learner$train(task, row_ids = 1:20)$predict(task, row_ids = 21:30)
 #' head(as.data.table(p))
+#' class(p$data$distr) # survival matrix stored
+#' p$distr # Matdist
```

Comment:
@RaphaelS1 => "But users aren't really meant to access the distribution via `$data$distr`, for them the main API is `$distr`. And by adding the comment it seems to imply that's always the case"

Reply:
willrevert change and add something more useful, as is shown in the book for example => https://mlr3book.mlr-org.com/chapters/chapter13/beyond_regression_and_classification.html#predict_type-distr

Reply:
My initial thoughts was to use the `$data$distr` (3d array) to draw the survival curves with uncertainty quantification for the mlr3 gallery post for BART. But since then I learned that a user can use `1 - distr6::gprm(pred$distr, 'cdf')` and get the same. Because `pred$distr$survival(times)` returns for a **specific curve** as we now have designed it.

Reply:
> will revert change and add something more useful, as is shown in the book for example => https://mlr3book.mlr-org.com/chapters/chapter13/beyond_regression_and_classification.html#predict_type-distr

I can't see anywhere here where we call `$data$distr`, this example calls `$distr` directly

> My initial thoughts was to use the $data$distr (3d array) to draw the survival curves with uncertainty quantification for the mlr3 gallery post for BART. But since then I learned that a user can use 1 - distr6::gprm(pred$distr, 'cdf') and get the full array. Can't do that with pred$distr$survival(times) since it returns for a specific curve as we now have designed it.

Internally you can (and should) definitely use `$data$distr`, my point above was just in documentation it isn't our primary external API

Reply:
Added all that in the doc now :)